vm/vmToX.go

```diff
@@ -51,18 +49,23 @@ func tryToBool(v reflect.Value, caseSensitive bool) (bool, error) {
 		return v.Bool(), nil
 	case reflect.String:
 		s := v.String()
-		if !caseSensitive {
-			s = strings.ToLower(s)
+		if v.Len() == 0 {
```

Comment:
I think the reflect.String can be simplified to just use ParseBool, remove all the extra stuff before and after it. Also please check the error from ParseBool and return if needed.

Reply:
ParseBool doesn’t work on “0.0” and also isn’t enough to validate whether the string is nonempty. An error from ParseBool means that the contents of the string couldn’t be converted to bool by Go specification but the string itself might still be non empty, contain a float, an integer not 0 or 1, etc. These errors aren’t really relevant to tryToBool which tries to get the Boolean value of the string itself rather than just its contents.

Reply:
Good point.
Not a big deal, but I think v.Len() == 0 can be moved up before s := v.String().